nilearn/decoding/decoder.py

```diff
@@ -119,14 +119,14 @@ def _check_param_grid(estimator, X, y, param_grid=None):
     if param_grid is None:
         param_grid = _default_param_grid(estimator, X, y)
 
-    elif isinstance(estimator, (RidgeCV, RidgeClassifierCV)):
+    elif isinstance(estimator, (RidgeCV, RidgeClassifierCV)) or (
+        not isinstance(estimator, LogisticRegressionCV)
+        and isinstance(estimator, LassoCV)
+    ):
```

Comment:
if we agree that those different options are mutually exclusive this can even be refactored even further

Reply:
I think this would try to set the `param_grid` for "alphas" with estimators that were not in the previous if.. elif cases.

I think this would happen for LinearSVC and SVR. For these estimators this would probably fail because they do not have a parameter alpha.

Reply:
However all the tests are passing, so either I am mistaken or this isn't getting caught in our tests

Reply:
OK I will investigate and see if something is missing in our tests or if we should have an else --> error to catch unknown cases